server/pbench/bin/index-pbench

```diff
@@ -187,99 +319,303 @@ def es_index(es, actions, dbg=0):
             break
     return 0
 
-###########################################################################
-# get metadata from metadata.log
-# metadata.log is in the format expected by configparser,
-# so we just get and return a configparser object.
-def get_mdconfig(mdf):
-
-    mdfconfig = SafeConfigParser()
-    mdfconfig.read(mdf)
-
-    # FIXME: verify start_run and end_run dictionary items are present
-    d = {}
-    d.update(mdfconfig.items('run'))
-    if "start_run" not in d or "end_run" not in d:
-        raise Exception('Missing start_run or end_run elements')
-    return mdfconfig
 
 ###########################################################################
-# generic metadata routines
-
-class UnsupportedTarballFormat(Exception):
-    pass
-
-
-def mk_pbench_metadata(mdconf):
-    d = {}
-    try:
-        d['pbench-rpm-version'] = mdconf.get("pbench", "rpm-version")
-    except Exception:
-        d['pbench-rpm-version'] = "Unknown"
-    return d
-
-def _to_utc(l, u):
-    """l is a timestamp in local time and u is one in UTC. These are
-    supposed to be close, so we calculate an offset rounded to a
-    half-hour and add it to the local date. We then convert back to an
-    ISO format date.
-
-    """
-    from datetime import datetime, timedelta
-
-    lts = datetime.strptime(l, "%Y-%m-%dT%H:%M:%S")
-    uts = datetime.strptime(u[:u.rfind('.')] , "%Y-%m-%dT%H:%M:%S")
-    offset = lts - uts
-    res = round(float(offset.seconds) / 60 / 60 + offset.days * 24, 1)
-    return (lts - timedelta(0, int(res*60*60), 0)).isoformat()
-
+# Tool data routines
+
+# Handlers data table / dictionary describing how to process a given tools
+# .csv files.  The outer dictionary holds one dictionary for each tool. Each
+# tool entry has one '@prospectus' entry, which records behavior and handling
+# controls, and then on entry for each .csv file name emitted by the tool.
+#
+# For each .csv file, we record the metric "class", and its name, "metric".
+# E.g. The metadata path hierarchy for iostat disk_IOPS.csv data would be,
+# 'iostat.disk.iops', for disk_Queue_Size.csv it would be 'iostat.disk.qsize',
+# etc.
+#
+# We record the metrics "display" name (human readable, or really what is
+# currently used by pbench today in generated html charts), and the "units"
+# for the metric, but neither field is currently indexed (FIXME!).
```

Comment:
I did not index these fields because it seemed silly to include the header with each tool.  Perhaps this data would be better suited in the pbench results JSON document?  Or a single JSON document that describes once the tool data.